### PR TITLE
MCR-3067 reuse JVM forks for junit tests

### PR DESCRIPTION
**Pull request by yagee-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3067).

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java [hunk 1]
```diff
@@ -155,6 +154,11 @@ public final class MCRURIResolver implements URIResolver {
      * Creates a new MCRURIResolver
      */
     private MCRURIResolver() {
+        reInit();
```
Comment (Mewel) [inline mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java]: move reInit() to static {} block and remove constructor.

File: mycore-base/src/main/java/org/mycore/common/events/MCRShutdownHandler.java [hunk 2]
```diff
@@ -159,6 +161,30 @@ void runClosables() {
         }
     }
 
+    /**
+     * only used to make {@link MCRShutdownHandler#resetCloseables()} testable
```
Comment (Mewel) [inline mycore-base/src/main/java/org/mycore/common/events/MCRShutdownHandler.java]: Javadoc comment is ambiguous. It should explain that this method (and the following) should only be called by a test framework and are not for applications to use.

**Review by Mewel — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java (hunk 1 above)
Comment (yagee-de) [reply mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java]: is fixed

File: mycore-base/src/main/java/org/mycore/common/events/MCRShutdownHandler.java (hunk 2 above)
Comment (yagee-de) [reply mycore-base/src/main/java/org/mycore/common/events/MCRShutdownHandler.java]: is fixed

**Review by Mewel — DISMISSED** [review]

**Comment by sebhofmann** [comment]
I think its hard or almost impossible to implement a "reset" of mycore. The configured instances in MCRConfiguration2#instanceHolder do not get resetted and there are many other place e.g. singletons, which do not get resetted. The Tests are now unstable and they get more unstable with this pr. Wouldn't it be a better approach to run the tests parallel to increase the build performance?

File: mycore-base/src/main/java/org/mycore/common/events/MCRShutdownHandler.java [hunk 3]
```diff
@@ -159,6 +161,38 @@ void runClosables() {
...
+
+    /**
+     * Resets the {@link Closeable} requests in the shutdown handler.
+     * This method checks if there is a previous state ({@link #mark()}) and, if present, resets the shuttingDown
+     * and requests fields to their values from the mark.
+     * The operation is performed under a write lock to ensure thread safety.
+     * <p>
+     * <b>Warning:</b> It should be used only in unit tests to test this class while keeping a clean state.
+     */
+    protected void resetCloseables() {
```
Comment (Mewel) [inline mycore-base/src/main/java/org/mycore/common/events/MCRShutdownHandler.java]: Should be inside of junit.

**Review by Mewel — CHANGES_REQUESTED** [review]

**Review by Mewel — DISMISSED** [review]

**Review by Mewel — APPROVED** [review]

**Review by sebhofmann — APPROVED** [review]

**Pull request merged**